R/plot.R

```diff
@@ -405,100 +368,62 @@ plot_heatmap <- function(scores,
 #' @export
 #'
 #' @examples
-#' example1 <- scoringutils::example_continuous
+#' library(ggplot2)
+#' library(magrittr)
 #'
-#' plot_predictions(
-#'   example1,
-#'   x = "target_end_date",
-#'   filter_truth = list(
-#'     'target_end_date <= "2021-07-22"',
-#'     'target_end_date > "2021-05-01"'
-#'   ),
-#'   filter_forecasts = list(
-#'     "model == 'EuroCOVIDhub-ensemble'",
-#'     'forecast_date == "2021-06-07"'
-#'   ),
-#'   facet_formula = location ~ target_type,
-#'   range = c(0, 50, 90, 95)
-#' )
+#' example_continuous %>%
+#'   filter_data (
+#'     what = "truth",
+#'     target_end_date <= "2021-07-22",
+#'     target_end_date > "2021-05-01"
+#'   ) %>%
+#'   filter_data (
+#'     what = "forecast",
+#'     model == 'EuroCOVIDhub-ensemble',
+#'     forecast_date == "2021-06-07"
+#'   ) %>%
+#'   plot_predictions (
+#'     x = "target_end_date",
+#'     by = c("target_type", "location"),
+#'     range = c(0, 50, 90, 95)
+#'   ) +
+#'   facet_wrap(~ location + target_type, scales = "free_y")
+#'
+
+
 plot_predictions <- function(data = NULL,
-                             forecasts = NULL,
-                             truth_data = NULL,
-                             merge_by = NULL,
+                             by = NULL,
                              x = "date",
-                             filter_truth = list(),
-                             filter_forecasts = list(),
-                             filter_both = list(),
-                             range = c(0, 50, 90),
-                             facet_formula = NULL,
-                             facet_wrap_or_grid = "facet_wrap",
-                             ncol = NULL,
-                             scales = "free_y",
-                             allow_truth_without_pred = FALSE,
-                             remove_from_truth = c("model", "forecaster", "quantile", "prediction", "sample", "interval")) {
+                             range = c(0, 50, 90)) {
 
   # preparations ---------------------------------------------------------------
   # check data argument is provided
-  if (is.null(data) && (is.null(truth_data) | is.null(forecasts))) {
-    stop("need arguments 'data' in function 'score()', or alternatively 'forecasts' and 'truth_data'")
-  }
-
   if (is.null(data)) {
-    data <- merge_pred_and_obs(forecasts, truth_data, by = merge_by, join = "full")
-    if (nrow(data) == 0) {
-      warning("After attempting to merge, only an empty data.table was left")
-      return(data)
-    }
+    stop("need arguments 'data' in function 'score()")
   }
 
   # split truth data and forecasts in order to apply different filtering
   truth_data <- data.table::as.data.table(data)[!is.na(true_value)]
   forecasts <- data.table::as.data.table(data)[!is.na(prediction)]
```

Comment:
I'll need to think about it but I don't know if we need to separate them now that that filtering step has been moved outside this function. It might be simpler & clearer to keep a single df.

Reply:
If you don't separate them then at least you will have a lot of "removing NA values" warnings. And potentially other things, where "other things" is the reason I did it like that in the first place, which I don't remember anymore :D

Reply:
I think I would go the other way and keep the user passing in forecasts and truth. Then use the filter_ family to get in the format needed here. For the truth data often it would just be recovering the data most users had prior to score.